### ISSUE TITLE
About some variable

### ISSUE THREAD

**Issue by starhiking** [jhb86253817/PIPNet#2]
https://github.com/jhb86253817/PIPNet/blob/77e01c85afe49dd9a212bbda8965562de9263a7e/experiments/WFLW/pip_32_16_60_r101_l2_l1_10_1_nb10.py#L19

What is the ```num_nb``` ?

**Comment by starhiking** [comment]
And what is the ```NRM``` in the regression picture

**Comment by jhb86253817** [comment]
> And what is the `NRM` in the regression picture

_num_nb_ is the number of neighbors; _NRM_ means Neighbor Regression Module, which is proposed in this paper.

**Comment by starhiking** [comment]
> > And what is the `NRM` in the regression picture
> 
> _num_nb_ is the number of neighbors, and it is a hyper-parameter of _NRM_; _NRM_ means Neighbor Regression Module, which is proposed in this paper.

Copy that, I have no idea that your second version paper is published!

**Comment by starhiking** [comment]
Good work!

**Comment by jhb86253817** [comment]
> Good work!

Thank you!

**Comment by starhiking** [comment]
Your method is very useful for applications. 
And I also have some views in the following.


The formula 3 and 6 in the paper remain a little puzzle.
In my opinion,  $$ \sum_{s^*_{ijk}=1} $$ is not suitable here, which needs to be combined with the code to understand.


Different with CornerNet, ExtremeNet, your offset map and its loss consider the only one grid, which don't care about the output from other areas .
I think your offset head is equal to a FC layer, even more  redundant information. 
Have you tried to use FC layer to regress the offset value? 

And your neighbor heads may be consuming  when inference.
Have you tried to add a constraint for the `pred_merge` and `pred` at training, and cut the neighbor heads at inference.

**Comment by jhb86253817** [comment]
> Your method is very useful for applications.
> And I also have some views in the following.
> 
> The formula 3 and 6 in the paper remain a little puzzle.
> In my opinion, $$ \sum_{s^*_{ijk}=1} $$ is not clear here, which needs to be combined with the source code to understand.
> 
> Different with CornerNet, ExtremeNet, your offset map and its loss consider the only one grid, which don't care about the output from other areas .
> I think your offset head is equal to a FC layer, even more redundant.
> Have you tried to use FC layer to regress the offset value? maybe faster.
> 
> And your neighbor heads may be consuming when inference.
> Have you consider to add a constraint for the `pred_merge` and `pred` at training, and cut the neighbor heads at inference.

Thank you for the suggestions. I will consider about them.
